.github/workflows/ci-pipeline.yml

```diff
@@ -15,4 +15,31 @@ jobs:
       - name: Build docker image for tests
         run: docker build --target test --tag todo-app:test .
       - name: Run tests from docker image
-        run: docker run todo-app:test 
\ No newline at end of file
+        run: docker run todo-app:test 
+  
+  publish:
+    name: Publish
+    runs-on: ubuntu-latest
+    needs: build
+    if: github.event_name == 'push' && github.ref == 'refs/heads/main'
+    steps:
+      - uses: actions/checkout@v2
+      - name: Log into dockerhun
+        run: echo ${{ secrets.DOCKERHUB_ACCESS_TOKEN }} | docker login -u schandarana --password-stdin
+      - name: Build production image
+        run: docker build --target production --tag schandarana/todo-app:$GITHUB_SHA --tag schandarana/todo-app:latest .
+      - name: Push image to dockerhub
+        run: docker push schandarana/todo-app --all-tags
+      - name: Update image tag 
+        run: docker tag schandarana/todo-app:latest registry.heroku.com/shiv-devops-course-starter/web
+      - name: Login to heroku container registry
+        env:
+          HEROKU_API_KEY: ${{ secrets.HEROKU_API_KEY }}
```

Comment:
Minor, but given this is duplicated for a couple of lines, it's worth noting that you can specify the environment variables using an identical format at the job level (i.e. `env` indented like `if` and `steps`) which lets you share the variables between individual steps without repeatedly adding it